### PR TITLE
GODRIVER-2448 update golang/x/net to latest

### PR DESCRIPTION
**Pull request by prestonvasquez**
[GODRIVER-2448](https://jira.mongodb.org/browse/GODRIVER-2448)

Update the `golang/x/net` go dependency to [latest](https://pkg.go.dev/golang.org/x/net) to help [prevent against denial of service attacks](https://app.snyk.io/org/mongodb-default/project/d1fd28ee-07d4-4de7-88ed-e20dc42a307c#issue-SNYK-GOLANG-GOLANGORGXNETHTTP2-2313688).

### Notes

```
[2022/06/27 15:43:37.914] go: golang.org/x/net@v0.0.0-20220624214902-1bab6f366d9e requires
[2022/06/27 15:43:37.914] 	golang.org/x/sys@v0.0.0-20220520151302-bc2c85ada10a: missing go.sum entry; to add it:
[2022/06/27 15:43:37.914] 	go mod download golang.org/x/sys
```

```
../../go/pkg/mod/github.com/xdg-go/stringprep@v1.0.2/profile.go:4:2: missing go.sum entry for module providing package golang.org/x/text/unicode/norm (imported by github.com/xdg-go/stringprep); to add:
	go get github.com/xdg-go/stringprep@v1.0.2
```

### CODE REVIEW

File: vendor/github.com/klauspost/compress/README.md [hunk 1]
```diff
@@ -0,0 +1,438 @@
+# compress
```
Comment (prestonvasquez) [inline vendor/github.com/klauspost/compress/README.md:1]: Are we wanting to suppress this README?  It is the result of `go mod vendor`. [The README is 6 years old](https://github.com/klauspost/compress/blame/master/README.md), so it feels like it was intentionally not added with [GODRIVER-2164](https://jira.mongodb.org/browse/GODRIVER-2164) ?

**Review by qingyang-hu — APPROVED** [review]
It looks good to me. I don't see a reason to exclude the README.

File: go.mod [hunk 2]
```diff
@@ -22,6 +22,7 @@ require (
 	github.com/xdg-go/stringprep v1.0.2
 	github.com/youmark/pkcs8 v0.0.0-20181117223130-1be2e3e5546d
 	golang.org/x/crypto v0.0.0-20201216223049-8b5274cf687f
+	golang.org/x/net v0.0.0-20220624214902-1bab6f366d9e // indirect
```
Comment (matthewdale) [inline go.mod:25]: Optional: Consider adding a comment that mentions we have this indirect dependency because it resolves CVE-2021-44716.

**Review by matthewdale — COMMENTED** [review]
Please run `go mod vendor` and commit the updated "vendor" directory.

**Review by qingyang-hu — APPROVED** [review]
👍

**Pull request merged**